sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt

```diff
@@ -214,6 +211,26 @@ object KlaviyoConfig : Config {
     }
 }
 
+internal fun PackageInfo.assertRequiredPermissions(requiredPermissions: Array<String>) {
+    val permissions = requestedPermissions?.toSet() ?: emptySet()
+    requiredPermissions.firstOrNull { it !in permissions }?.let { throw MissingPermission(it) }
+}
+
+internal fun buildUserAgent(context: Context, packageInfo: PackageInfo): String {
+    val versionCode = packageInfo.getVersionCode()
+    val applicationName = context.packageManager.getApplicationLabel(context.applicationInfo)
+
+    return "$applicationName/${packageInfo.versionName} (${packageInfo.packageName}; build:$versionCode; Android ${Build.VERSION.SDK_INT}) klaviyo-android/${BuildConfig.VERSION}"
```

Comment:
super nit, but is there a way to break this down so it isn't a super long line? Like maybe
```
"%s/%s (%s; build:%s ...etc".format(
   applicationName,
   versionName,
   etc...
 )
```
IDK which is more readable anyways.